Dockerfile

```diff
@@ -27,37 +29,117 @@ LABEL IMAGE_SOURCE="https://github.com/ArkCase/ark_jenkins_build"
 ENV APP_USER="${APP_USER}"
 ENV APP_UID="${APP_UID}"
 ENV APP_GID="${APP_GID}"
+ENV NVM_VER="${NVM_VER}"
 
 #
 # O/S updates, and base tools
 #
-RUN apt-get update && apt-get -y dist-upgrade
-RUN apt-get install -y git sudo make gcc openssl
+RUN apt-get update && \
+	apt-get -y dist-upgrade -f && \
+	apt-get install -y \
+		autoconf \
+		automake \
+		bzip2 \
+		bzr \
+		ca-certificates \
+		curl \
+		dirmngr \
+		default-libmysqlclient-dev \
+		dpkg-dev \
+		file \
+		g++ \
+		gcc \
+		gcc \
+		git \
+		gnupg \
+		imagemagick \
+		libbz2-dev \
+		libc6-dev \
+		libcurl4-openssl-dev \
+		libdb-dev \
+		libevent-dev \
+		libffi-dev \
+		libgdbm-dev \
+		libgeoip-dev \
+		libglib2.0-dev \
+		libgmp-dev \
+		libjpeg-dev \
+		libkrb5-dev \
+		liblzma-dev \
+		libmagickcore-dev \
+		libmagickwand-dev \
+		libncurses5-dev \
+		libncursesw5-dev \
+		libpng-dev \
+		libpq-dev \
+		libreadline-dev \
+		libsqlite3-dev \
+		libssl-dev \
+		libtool \
+		libwebp-dev \
+		libxml2-dev \
+		libxslt-dev \
+		libyaml-dev \
+		make \
+		make \
+		mercurial \
+		netbase \
+		openssh-client \
+		openssl \
+		patch \
+		procps \
+		sshpass \
+		subversion \
+		sudo \
+		unzip \
+		vim \
+		wget \
+		xz-utils \
+		zip \
+		zlib1g-dev \
+	&& apt-get clean && rm -rf /var/lib/apt/lists/*
 
 #
 # Create the user and their home
 #
+RUN groupadd --system "build"
 RUN groupadd --gid "${APP_GID}" "${APP_GROUP}"
-RUN useradd --uid "${APP_UID}" --gid "${APP_GID}" -m --home-dir "/home/${APP_USER}" "${APP_USER}"
+RUN useradd --uid "${APP_UID}" --gid "${APP_GID}" --groups "build" -m --home-dir "/home/${APP_USER}" "${APP_USER}"
 
 #
-# We use a relative paths for the links b/c this makes them "constant"
+# Add the entrypoint
 #
-RUN ln -s ../../cache "/home/${APP_USER}"
-RUN ln -s ../../tools "/home/${APP_USER}"
+COPY --chown=root:root entrypoint /
+RUN chmod 0755 /entrypoint
 
 #
-# Add the entrypoint
+# Add the sudo configuration for the build group
+#
+COPY --chown=root:root 00-build /etc/sudoers.d
+RUN chmod 0640 /etc/sudoers.d/00-build
+
+#
+# Now do the configurations for the actual user
+#
+USER "${APP_USER}"
+
+#
+# Disable SSL certificate verification for GIT (for now ... this probably needs to go away)
```

Comment:
Right, we could use openssl to lift the root certificate from gitlab.armedia.com and add it to the image, so it becomes a trusted site.

Reply:
Yeah I'm moving this to a script in init.d so we can turn it on/off without messing with the image